build.go

```diff
@@ -25,12 +32,33 @@ func Build(
 			return packit.BuildResult{}, err
 		}
 
+		sum, err := calculator.Sum(filepath.Join(context.WorkingDir, "Gemfile"))
```

Comment:
I don't think this can be consistently trusted to change to force a cache invalidation. Instead, I think you should checksum the `Gemfile.lock` iff it is present.

The problem with using the `Gemfile` is that I can include a directive like `gem 'thin',  '~>1.1'` and have `bundle-install` install the `1.1.1` version of `thin` in a first build. Then, the `thin` maintainers could release a `1.1.2` version, after which I may build my app again, with no substantive changes to the `Gemfile`. In that case, I would expect `bundle-install` to run again and install `thin 1.1.2`.

Forcing the app developer to make a change to the Gemfile just to get the cache to invalidate would be painful.